src/spu-schema/src/server/stream_fetch.rs

```diff
@@ -86,10 +95,53 @@ impl Default for SmartStreamKind {
 #[derive(Debug, Clone, Encoder, Decoder)]
 pub enum SmartStreamWasm {
     Raw(Vec<u8>),
+    /// compressed WASM module payload using Gzip
+    #[fluvio(min_version = 14)]
+    Gzip(Vec<u8>),
     // TODO implement named WASM modules once we have a WASM store
     // Url(String),
 }
 
+fn zip(raw: Vec<u8>) -> io::Result<Vec<u8>> {
+    let mut encoder = GzEncoder::new(raw.as_ref(), Compression::default());
+    let mut buffer = Vec::with_capacity(raw.len());
+    encoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+fn unzip(compressed: Vec<u8>) -> io::Result<Vec<u8>> {
+    let mut decoder = GzDecoder::new(compressed.as_ref());
+    let mut buffer = Vec::with_capacity(compressed.len());
+    decoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+impl SmartStreamWasm {
+    /// returns the gzip-compressed WASM module bytes
+    pub fn to_gzip(self) -> io::Result<Self> {
+        Ok(match self {
+            Self::Raw(raw) => Self::Gzip(zip(raw)?),
+            Self::Gzip(_) => self,
+        })
+    }
```

Comment:
Where these are used, it seems like this method is just being used to re-assign a `SmartStreamWasm` value, i.e. above where I commented, you have

```rust
module.wasm = module.wasm.to_gzip()?;
```

Since that's the usage pattern, it would probably make sense to adjust `to_gzip` and `to_raw` to look more like this:

```rust
pub fn to_gzip(&mut self) -> io::Result<()> {
    *self = match self {
        Self::Raw(raw) => Self::Gzip(zip(raw)?),
        Self::Gzip(_) => self,
    };
    Ok(())
}
```

Which would let you change the caller to look like:

```rust
module.wasm.to_gzip()?;
```